Assets/Scripts/Common/Effects/DamageEffect.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DamageEffect : MonoBehaviour {
+	[SerializeField] private ParticleSystem damageEffect;
+
+	private void Start() {
+		GetComponent<DamageArea>().OnDamageEffect.AddListener(Spawn);
```

Comment:
Will likely never be a problem, but the fact that this listener is never removed makes me anxious.